src/wp-includes/block-supports/typography.php

```diff
@@ -98,119 +95,358 @@ function wp_apply_typography_support( $block_type, $block_attributes ) {
+		$value = $value / $options['root_size_value'];
+		$unit  = $options['coerce_to'];
+	}
+
+	return array(
+		'value' => $value,
+		'unit'  => $unit,
+	);
+}
+
+/**
+ * Internal implementation of clamp() based on available min/max viewport width, and min/max font sizes.
+ *
+ * @since 6.1.0
+ * @access private
+ *
+ * @param array $args {
+ *     Optional. An associative array of values to calculate a fluid formula for font size. Default is empty array.
+ *
+ *     @type string $maximum_viewport_width Maximum size up to which type will have fluidity.
+ *     @type string $minimum_viewport_width Minimum viewport size from which type will have fluidity.
+ *     @type string $maximum_font_size      Maximum font size for any clamp() calculation.
+ *     @type string $minimum_font_size      Minimum font size for any clamp() calculation.
+ *     @type int    $scale_factor           A scale factor to determine how fast a font scales within boundaries.
+ * }
+ * @return string|null A font-size value using clamp().
+ */
+function wp_get_computed_fluid_typography_value( $args = array() ) {
+	$maximum_viewport_width_raw = isset( $args['maximum_viewport_width'] ) ? $args['maximum_viewport_width'] : null;
+	$minimum_viewport_width_raw = isset( $args['minimum_viewport_width'] ) ? $args['minimum_viewport_width'] : null;
+	$maximum_font_size_raw      = isset( $args['maximum_font_size'] ) ? $args['maximum_font_size'] : null;
+	$minimum_font_size_raw      = isset( $args['minimum_font_size'] ) ? $args['minimum_font_size'] : null;
+	$scale_factor               = isset( $args['scale_factor'] ) ? $args['scale_factor'] : null;
+
+	// Grab the minimum font size and normalize it in order to use the value for calculations.
+	$minimum_font_size = wp_get_typography_value_and_unit( $minimum_font_size_raw );
+
+	// We get a 'preferred' unit to keep units consistent when calculating, otherwise the result will not be accurate.
+	$font_size_unit = isset( $minimum_font_size['unit'] ) ? $minimum_font_size['unit'] : 'rem';
+
+	// Grab the maximum font size and normalize it in order to use the value for calculations.
+	$maximum_font_size = wp_get_typography_value_and_unit(
+		$maximum_font_size_raw,
+		array(
+			'coerce_to' => $font_size_unit,
+		)
+	);
+
+	// Protect against unsupported units.
+	if ( ! $maximum_font_size || ! $minimum_font_size ) {
+		return null;
+	}
+
+	// Use rem for accessible fluid target font scaling.
+	$minimum_font_size_rem = wp_get_typography_value_and_unit(
+		$minimum_font_size_raw,
+		array(
+			'coerce_to' => 'rem',
+		)
+	);
+
+	// Viewport widths defined for fluid typography. Normalize units.
+	$maximum_viewport_width = wp_get_typography_value_and_unit(
+		$maximum_viewport_width_raw,
+		array(
+			'coerce_to' => $font_size_unit,
+		)
+	);
+	$minimum_viewport_width = wp_get_typography_value_and_unit(
+		$minimum_viewport_width_raw,
+		array(
+			'coerce_to' => $font_size_unit,
+		)
+	);
```

Comment:
Why is this needed twice? @ockham

Reply:
TBH I know next to nothing about this code -- I've only rebased it 😅 

Speaking of rebase, maybe the duplicated code is a result of a slightly wonky rebase?

Reply:
> Nope, it's the original Gutenberg PR https://github.com/WordPress/gutenberg/pull/39529.

Ah, I hadn't noticed! Huh, weird.

(If it's blocking this from merging, I'd go out on a limb and say that we might remove it? 😅 )

Reply:
Hmm, it's been duplicated for a while: https://github.com/WordPress/gutenberg/commit/eea3aebb1ab4de6fa300bd24bfb1f7afa3b8051e#diff-2c54d3c8305590cf826f83511a9fd85d5b405470addd62edc3183ff9118177fbL302

Reply:
I'm going to remove it. If it's an issue, a Trac ticket can be opened to fix it during the beta cycle.

Reply:
Scratch that - my bad 🤦‍♀️ The 2nd time is for the minimum. Sorry for the confusion. Leaving it in (and smacking my forehead 🤣 ).

Reply:
Thanks for the ping, folks!

Yep, one is for `$minimum_viewport_width` and the other for `$maximum_viewport_width`. 

If you think the vars could be named more clearly feel free to change. I can sync in Gutenberg today.